packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -341,17 +341,31 @@ class CupertinoTextField extends StatefulWidget {
        ),
        assert(enableIMEPersonalizedLearning != null),
        keyboardType = keyboardType ?? (maxLines == 1 ? TextInputType.text : TextInputType.multiline),
-       toolbarOptions = toolbarOptions ?? (obscureText ?
-         const ToolbarOptions(
-           selectAll: true,
-           paste: true,
-         ) :
-         const ToolbarOptions(
-           copy: true,
-           cut: true,
-           selectAll: true,
-           paste: true,
-         )),
+       enableInteractiveSelection = enableInteractiveSelection ?? (!readOnly || !obscureText),
+       toolbarOptions = toolbarOptions ??
```

Comment:
Trying to brainstorm ideas for this code duplication still...  Could you pass it directly to EditableText without any initializer, and do this logic only in EditableText?

Reply:
Hmm. Maybe. I'll see if that works.  If it does, then I can probably remove the duplicate tests too.

Reply:
Well, the only way I can do that is if I make `TextField.toolbarOptions` nullable, or the constructor argument required, either one of which is a breaking change. I think I prefer the duplicated code to the breaking change (although I'm not happy about it).